drivers/sdp3x/sdp3x.c

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright (C) 2019 Dirk Ehmen
+ *               2020 Jan Schlichter
+ *               2020 Nishchay Agrawal
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sdp3x
+ * @{
+ * @file
+ * @brief       Sensirion SDP3x sensor driver implementation
+ *
+ * @author      Dirk Ehmen       <ehmen@ibr.cs.tu-bs.de>
+ * @author      Jan Schlichter   <schlichter@ibr.cs.tu-bs.de>
+ * @author      Nishchay Agrawal <f2016088@pilani.bits-pilani.ac.in>
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include "sdp3x_params.h"
+#include "sdp3x.h"
+#include "checksum/crc8.h"
+#include "thread.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define DEV_I2C      (dev->params.i2c_dev)
+#define DEV_ADDR     (dev->params.i2c_addr)
+#define DEV_MODEL    (dev->params.model)
+
+static int8_t _checkCRC(uint16_t value, uint8_t test);
+static int8_t _SDP3x_read_data(const sdp3x_t *dev, int16_t *data);
+static double _SDP3x_convert_to_pascal(const sdp3x_t *dev, int16_t value);
+static double _SDP3x_convert_to_inches_h2o(const sdp3x_t *dev,
+                                           int16_t value) __attribute__(
+    (unused));
+static double _SDP3x_convert_to_celsius(int16_t value);
+int8_t _SDP3x_start_triggered(const sdp3x_t *dev, uint8_t tempComp,
+                              uint8_t clkStretching);
+double _SDP3x_read_temp(const sdp3x_t *dev);
+double _SDP3x_read_pressure(const sdp3x_t *dev);
+
+int SDP3x_init(sdp3x_t *dev, const sdp3x_params_t *params)
+{
+    DEBUG("[SDP3x] init: Initializing device\n");
+
+    dev->params = *params;
+
+    i2c_init(DEV_I2C);
+    SDP3x_soft_reset(dev);
+
+    /* try to read product number to check if sensor is connected and working */
+    int ret = 0;
+    uint8_t cmd1[2] = { 0x36, 0x7C };
+    uint8_t cmd2[2] = { 0xE1, 0x02 };
+    i2c_acquire(DEV_I2C);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, cmd1, 2, I2C_NOSTOP);
+    if (ret != 0) {
+        i2c_release(DEV_I2C);
+        return ret;
+    }
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, cmd2, 2, I2C_NOSTOP);
+    if (ret != 0) {
+        i2c_release(DEV_I2C);
+        return ret;
+    }
+    i2c_release(I2C_DEV(0));
+
+#ifdef SDP3X_PARAM_IRQ_PIN
+    gpio_init(params->irq_pin, GPIO_IN);
+#endif
+
+    DEBUG("[SDP3x] init: Init done\n");
+    return 1;
+}
+
+double SDP3x_read_single_temperature(const sdp3x_t *dev, uint8_t tempComp,
+                                     uint8_t clkStretching)
+{
+    _SDP3x_start_triggered(dev, tempComp, clkStretching);
+#ifndef SDP3X_PARAM_IRQ_PIN
+    /* Wait for measurement to be ready if irq pin not used */
+    xtimer_usleep(46000);
```

Comment:
46000 should be a `define`.

Reply:
Resolved in fixup commit 6b544ab8d1ee03eeff471b3d4190e00be1f6e85f